Library/Formula/postgresql.rb

```diff
@@ -77,8 +82,16 @@ def install
   end
 
   def post_install
-    unless File.exist? "#{var}/postgres"
-      system "#{bin}/initdb", "#{var}/postgres"
+    system "pkill postgres || true"
+
+    if (var/"postgres").exist?
+      mv var/"postgres", var/"postgres-#{version_suffix}"
```

Comment:
Apologies, I missed this before I hit the confirm on ` git push `. I think 1 & 3 are handled in the new commit. Let me know how you feel about 2. I guess we could do something like:

```
ENV["PGDATAOLD"] = #{HOMEBREW_CELLAR}/postgresql/oldversion/share/name
ENV["PGDATANEW"] = #{Formula["postgresql"].opt_share}/name
ENV["PGBINOLD"] = #{HOMEBREW_CELLAR}/postgresql/oldversion/bin
ENV["PGBINNEW" = #{Formula["postgresql"].opt_bin}
system "#{bin}/pg_upgrade"
```

?